car_cleaning_service_with_threads.rb

```diff
@@ -0,0 +1,115 @@
+require'time'
+
+class Car
+  attr_reader :id
+
+  def initialize(id)
+    @id = id
+  end
+
+  def to_s
+    @id
+  end
+
+  def to_sym
+    @id.to_sym
+  end
+end
+
+class CarCleaningService
+
+  MAX_CARS_AT_ONCE = 2
+  CAR_CLEANING_TIME = 2
+  ONE_HOUR_IN_SECONDS = 3600
+
+  attr_reader :time
+
+  def initialize(queue = Queue.new)
```

Comment:
what if you pass the Car object here in the initializer ? In this way you will get rid of the car parameter passed to the methods.

Reply:
I will reformat the code such that the CarCleaningService will be split into a model (the CarCleaner itself, holding only the data) and a service (CarCleanerService, implementing the functionality related to the CarCleaner) and I cannot pass the Car object itself to the CarCleaner as it holds a list of cars that enter this facility, and initially it is empty. As cars arrive, they are inserted into the queue based on the time they are needed (the queue is an array which behaves like a priority queue).

Reply:
Awesome, let's see how it will look like afterwards.